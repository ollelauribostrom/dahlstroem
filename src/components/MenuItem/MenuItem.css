.menu-item {
  color: #151515;
  cursor: pointer;
  margin-left: 30px;
}

.menu-item--active {
  border-bottom: 1px solid #151515;
}

.menu-item__link {
  color: #151515;
}
