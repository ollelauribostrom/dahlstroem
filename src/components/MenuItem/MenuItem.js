import PropTypes from 'prop-types';
import React from 'react';
import { Link } from 'gatsby';

const MenuItem = ({ title, url, isActive }) => (
  <span
    tabIndex={0}
    className={`menu-item ${isActive ? 'menu-item--active' : ''}`}
  >
    <Link to={url} className="menu-item__link">
      {title}
    </Link>
  </span>
);

MenuItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  isActive: PropTypes.bool,
};

MenuItem.defaultProps = {
  isActive: false,
};

export default MenuItem;
