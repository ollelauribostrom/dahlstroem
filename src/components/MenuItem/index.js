import MenuItem from './MenuItem';
import './MenuItem.css';

export default MenuItem;
