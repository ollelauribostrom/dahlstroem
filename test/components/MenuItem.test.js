import React from 'react';
import { shallow } from 'enzyme';

import MenuItem from '../../src/components/MenuItem';

describe('Tests for MenuItem', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <MenuItem color={{ hex: '#fff' }} onMouseOver={() => {}}>
        <span>child</span>
      </MenuItem>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('renders correctly when active', () => {
    const wrapper = shallow(
      <MenuItem isActive={true} onMouseOver={() => {}}>
        <span>child</span>
      </MenuItem>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
