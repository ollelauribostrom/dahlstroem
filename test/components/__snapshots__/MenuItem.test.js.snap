// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for MenuItem renders correctly 1`] = `
<span
  onMouseOver={[Function]}
  style={
    Object {
      "borderBottom": "none",
      "color": "#fff",
      "cursor": "pointer",
    }
  }
  tabIndex={0}
>
  <span>
    child
  </span>
</span>
`;

exports[`Tests for MenuItem renders correctly when active 1`] = `
<span
  onMouseOver={[Function]}
  style={
    Object {
      "borderBottom": "1px solid #000",
      "color": "#000",
      "cursor": "pointer",
    }
  }
  tabIndex={0}
>
  <span>
    child
  </span>
</span>
`;
